pkg/auth/multiauthenticator.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+type IssuerWithClientID struct {
+	Issuer   string // Issuer of the token.
+	ClientID string // ClientID of the token, can be empty.
```

Comment:
maybe add a comment to indicate what it means to have an empty ClientID. This was not obvious to me.

Reply:
I have removed the comment and added an explicit check to fail if clientID is found to be empty. This behavior is clearer as compared to allowing empty clientIDs. Another thought is this multi authenticator really makes sense only for OIDC authenticators where we definitely need a clientID to verify the token. May be we could rename this to OIDCMultiAuthenticatorsWithClientID or so